mali: support building against 4.17

4.17 changed the return code of the fault handler from int to vm_fault_t.

The return values don't change though, so we only need to update the prototype.

Signed-off-by: Maxime Ripard <maxime.ripard@bootlin.com>

--- a/src/devicedrv/mali/linux/mali_memory.c
+++ b/src/devicedrv/mali/linux/mali_memory.c
@@ -70,7 +70,9 @@
 	}
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
+static vm_fault_t mali_mem_vma_fault(struct vm_fault *vmf)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
 static int mali_mem_vma_fault(struct vm_fault *vmf)
 #else
 static int mali_mem_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
